jakarta-eclipse/org.eclipse.lsp4jakarta.core/src/org/jakarta/codeAction/proposal/ModifyAnnotationProposal.java

```diff
@@ -138,9 +158,82 @@ protected ASTRewrite getRewrite() throws CoreException {
+                            StringLiteral stringValue = ast.newStringLiteral();
+                            stringValue.setLiteralValue("");
+                            memberValuePair.setValue(stringValue);
+                            values.add(memberValuePair);
+                        }
+                    }
+                    
+                    rewrite.getListRewrite(declNode,
+                            isField ? FieldDeclaration.MODIFIERS2_PROPERTY : TypeDeclaration.MODIFIERS2_PROPERTY)
+                            .insertFirst(marker, null);
+                }    
+            }
+            
+            return rewrite;
+        } else if (declNode instanceof TypeDeclaration || isField) {
+            AST ast = declNode.getAST();
+            ASTRewrite rewrite = ASTRewrite.create(ast);
+            
+            ImportRewriteContext importRewriteContext = new ContextSensitiveImportRewriteContext(declNode, imports);
+            List<Annotation> existingAnnotations = new ArrayList<Annotation>();
+            List<? extends ASTNode> children = (List<? extends ASTNode>) declNode
+            		.getStructuralProperty(TypeDeclaration.MODIFIERS2_PROPERTY);
 
+            // find and save existing annotation, then remove it from ast
+            for (ASTNode child : children) {
+                if (child instanceof Annotation) {
+                    Annotation annotation = (Annotation) child;
+                    boolean containsAnnotation = Arrays.stream(annotationShortNames)
+                            .anyMatch(annotation.getTypeName().toString()::contains);
+                    if (containsAnnotation) {
+                        existingAnnotations.add(annotation);
+                        rewrite.remove(child, null);
                     }
+                }
+            }
+           
+            // add new annotation with fields from existing annotation
+            for (String annotation : annotations) {
+                NormalAnnotation marker = ast.newNormalAnnotation();
+                marker.setTypeName(ast.newName(imports.addImport(annotation, importRewriteContext)));
+                List<MemberValuePair> values = marker.values();
+                
+                if (!existingAnnotations.isEmpty()) {
+                	for (Annotation a : existingAnnotations) {
+                	    if (a instanceof NormalAnnotation) {
+                	        List<MemberValuePair> existingValues = ((NormalAnnotation) a).values();
+                            for (MemberValuePair mvp : existingValues) {
+                            	boolean removeAttribute = this.attributesToRemove
+                            			.contains(mvp.getName().getFullyQualifiedName());
 
+                                 // do not add attribute to be removed
+                            	if (!removeAttribute) {
+                            		MemberValuePair memberValuePair = ast.newMemberValuePair();
```

Comment:
Looks like a tab is here - use 4 spaces instead of tabs

Reply:
921d26e